drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(power_domain_intel_adsp, LOG_LEVEL_INF);
+
+struct pg_registers {
+	uint32_t wr_address;
+	uint32_t rd_address;
+	uint32_t SPA_bit;
+	uint32_t CPA_bit;
+};
+
+static int pd_intel_adsp_set_power_enable(struct pg_registers *reg, bool power_enable)
```

Comment:
This looks to be doing exactly the same thing as the code in mp_cavs.c.  Can we please unify?  We don't want two implementations in-tree.  Maybe use a shared soc-level utility that they both call, delete the old one if it's no longer needed, etc...

Reply:
That is a good point indeed.

Reply:
Yes, the mechanism is similar. It seems like in cavs it's used only to block core gating and there we're using it in a broader scope. Cavs part may be redesigned to work with this implementation, but it'd need a bit of tinkering to do that in a clean way (mostly redefining register structure where PWRCTL resides and moving it to devicetree). I don't have an easy access to cavs documentation and this is targeting ace primarily anyway, so I'm not willing to do that, at the moment at least, if not necessary.